[s] Highlander 2.0

Xhuis opened the pull request:
:cl: Xhuis
add: Highlander has been revamped. Now you can butcher your coworkers in cold blood like never before!
/:cl:
- Activating Highlander now plays a global sound filled with [DRUMS AND BAGPIPES](https://dl.dropboxusercontent.com/u/116194515/highlander.ogg), as it always should have. The message has also been tweaked, and the shuttle is immediately called. Finally, all humans lose the ability to use guns of any kind.
- Kilts, berets, and claymores given by Highlander are now nodrop.
- Highlander now grants unique claymores. Slaying an enemy with your claymore will heal all brute damage and scratch a notch into the blade for each kill. Your blade grows more powerful as you slay with it, gaining 1 point of damage for every kill and becoming legendary at 10 kills.
  - These claymores also have a 100% chance to deflect energy projectiles and make you immune to stuns.
  - If you die while holding one, you are subsequently gibbed. Your brain is also deleted.
  - Killing someone with a claymore steals their soul and traps it into the blade. They can talk from here but nothing else, and die along with the claymore should it somehow be destroyed.
  - These claymores can't block melee attacks.
- The last man standing will gain special mention across the entire server and bagpipe fanfare.

Commit:
Highlander 2.0

Commit:
Adds a return

PKPenguin321 commented:
SSSSSCOTLAAAAAAAAAAAAAAAAAAAAAAAAAAAAAND FOREEEEEEEEVERRRRRRRRRRRRRRRRRRRRRRRRR

optimumtact commented:
LADS FOR SCOTLAND

yackemflam commented:
Did you make it so that if anyone uses a gun during highlander they gib?

Xhuis commented:
@yackemflam No, but the claymores deflect everything.

TechnoAlchemisto commented:
Does this mean that riot shotguns and detective pistols will not deflect since they are not energy

Xhuis commented:
I'm going to test that in just a minute. If they don't deflect, I'll make it so that they will.

Commit:
Soul stealing and anti-guns

Xhuis commented:
* Added soul stealing. Killing someone with a claymore traps their soul in the sword, giving them a front-row seat to the carnage.
* Claymores can no longer block melee hits. RNG won't decide these fights now.
* Made the attack verbs for claymores more visceral.
* People gibbed by claymores lose their brains.
* Activating Highlander now removes the ability for anyone to use guns.

Commit:
Anti-baton

Xhuis commented:
* Claymores instantly get you back up from stuns every tick.

Ergovisavi commented:
> Claymores can no longer block melee hits. RNG won't decide these fights now.

Can't you still get RNG weakened with a chest hit, or RNG paralyzed with a blow to the head?

Supermichael777 commented:
add all scotsmen to a list.
if your the last on the list you become the one true Scotsman and immortal

nullbear commented:
Ah, one more thing you should keep note of.

When you join a round mid-match, you'll be a normal crewmember. Change this?

Commit:
Announces the last man standing

Xhuis commented:
* Being the last man standing now announces your name to the entire server (complete with inspiring bagpipe fanfare) and removes the NODROP flag on your claymore.

Cheridan commented:
I'd like if it just deleted all guns when pressed, or at least turned them into bows or something.

Alek2ander commented:
Instread of gibbing it should decapitate.

xxalpha commented:
Why is this allowed to have a silent tag?

X-TheDark commented on code/game/objects/items/weapons/weaponry.dm:
```diff
@@ -54,6 +54,116 @@
+			for(var/obj/item/weapon/claymore/highlander/H in highlander_claymores)
+				if(H != src)
+					return
+			announced = TRUE
```
Are you sure this won't trigger erroneously?
If all claymores are in a list, one of them is bound to find themselves in the first position of the said list...

KorPhaeron commented:
Give all the highlanders sleeping carp so they can't shoot or be shot

Xhuis commented:
@xxalpha I didn't want to spoil it for the masses.
@KorPhaeron I considered that, but I decided that the combo attacks might be too powerful.

Fox-McCloud commented on code/game/objects/items/weapons/weaponry.dm:
```diff
@@ -54,6 +54,116 @@
+			L << "<span class='userdanger'>You are unworthy.</span>"
+			L.visible_message("<span class='warning'>[L] explodes in a shower of gore!</span>")
+			var/obj/item/organ/brain/B = L.getorgan(/obj/item/organ/brain)
+			L.gib()
```
There's a specific argument for gib that I made that ensures the brain isn't ejected, and therefore destroyed when it gibs---Unless that was removed, that would be better to use.

Fox-McCloud commented on code/game/objects/items/weapons/weaponry.dm:
```diff
@@ -54,6 +54,116 @@
+			world << "<span class='userdanger'>[L.real_name] IS THE ONLY ONE LEFT STANDING!</span>"
+			world << sound('sound/misc/highlander_only_one.ogg')
+			L << "<span class='notice'>YOU ARE THE ONLY ONE LEFT STANDING! You feel your bonds vanish - you can freely drop or sheathe your [name]! (The kilt stays, though.)</span>"
+			flags = CONDUCT
```
Toggle the NODROP flag off, that's better practice than setting the flag directly.

Fox-McCloud commented on code/game/objects/items/weapons/weaponry.dm:
```diff
@@ -54,6 +54,116 @@
+
+/obj/item/weapon/claymore/highlander/proc/add_notch(mob/living/user)
+	notches++
+	force++
```
I'd recommend higher force than just 1, personally, but that's me.

Xhuis replied:
I don't want to give a _huge_ advantage, so that everyone has a chance to get a kill or two before they get cucked.

kevinz000 commented:
i want a strange professional killstreak claymore now
^_^

Davdichan commented:
Should  have someone sprite bagpipes and make them function like clown horn in that they play noise to everyone on screen and then some, Scotties can run around blowing their pipes to attract new challengers.

Alek2ander commented:
More lore: you should prevent attacks when inside the chapel.

xxalpha commented:
We need bagpipe instrument.

KingDairy commented on code/game/objects/items/weapons/weaponry.dm:
```diff
@@ -54,6 +54,116 @@
+	block_chance = 0 //RNG WON'T HELP YOU NOW, PANSY
+	attack_verb = list("brutalized", "eviscerated", "disemboweled", "hacked", "carved", "cleaved", "gored") //ONLY THE MOST VISCERAL ATTACK VERBS
+	var/notches = 0 //HOW MANY PEOPLE HAVE BEEN SLAIN WITH THIS BLADE
+	var/announced = FALSE //If we've announced our winner
```
IT'S NOT IN CAPS
YOU BROKE THE RULES

Commit:
Adjustments based on feedback

Xhuis commented:
* Made claymores grant stun immunity instead of just cleansing them every tick.
* Code cleanup and improvements.

optimumtact commented on code/modules/admin/verbs/onlyone.dm:
```diff
@@ -48,6 +47,10 @@
 
+		var/obj/item/weapon/claymore/highlander/H1 = new(H)
+		H.put_in_hands(H1)
+		H1.pickup(H)
```
The claymore is picking up the Human?

Xhuis replied:
Items have pickup as a proc that refers to the mob picking them up. It's called manually because Byond.

optimumtact commented on code/game/objects/items/weapons/weaponry.dm:
```diff
@@ -54,6 +54,118 @@
+		else
+			if(announced)
+				return
+			for(var/obj/item/weapon/claymore/highlander/H in highlander_claymores)
```
first of all, wouldn't it be easier to increment/decrement a highlander_count variable instead.

secondly, if only one highlander claymore spawns and processes before any others can spawn, won't that person win?

Xhuis replied:
1. I don't see why not
2. Yes, but it would probably never happen so I didn't put in a failsafe. Highlander is an admin button and I doubt they'd press it without a lot of people.

Jordie0608 replied:
They might just spawn a claymore outside of the event however.

Xhuis replied:
True, I'll make an override for it

Iamgoofball commented on code/game/objects/items/weapons/weaponry.dm:
```diff
@@ -54,6 +54,118 @@
+/obj/item/weapon/claymore/highlander/IsReflect()
+	return 1 //YOU THINK YOUR PUNY LASERS CAN STOP ME?
+
+/obj/item/weapon/claymore/highlander/proc/add_notch(mob/living/user) //DYNAMIC CLAYMORE PROGRESSION SYSTEMS - THIS IS THE FUTURE
```
can we procedurally generate the claymore too?

ExitGame commented:
please make scotland the brave play on loop

Commit:
Prevents admin-spawned claymores from announcing

Xhuis commented:
* Admin-spawned claymores no longer trigger announcements.

Xhuis commented:
Oh damn, I forgot to push a commit. I'll wrap up work on it and send it your way, but once that's here it's good.

Commit:
Removes some unneeded features, adds a coward-compass

Commit:
Removes nodrop from claymores

Xhuis commented:
* Claymores now have a coward compass. Activating the claymore will show the direction of the nearest fighter. This is designed to allow for constant action, as you always know where another enemy is.
* Removed soul claiming from claymores.
* Removed gibbing from claymores. Instead, killing someone with a claymore will dust the target.
* Removed nodrop from claymores.

TheDZD commented:
You might have issues with claymores not being NODROP, since if somebody drops theirs you'll never end up with only a single claymore left unless someone else picks it up and gets dusted. Additionally, if somebody dies due to being in crit rather than the claymore, you have the same issue.

Xhuis commented:
I primarily removed the flag since pressing E equipped them to the back or belt slot with no way to remove them. Since I didn't want to refactor equipping code, nor did I know where it was stored, I just sliced off the whole NODROP bit.

Iamgoofball commented:
>Removed gibbing from claymores. Instead, killing someone with a claymore will dust the target.

:-1: not metal enough

TechnoAlchemisto commented:
killing should probably behead more than anything

Xhuis commented:
@Iamgoofball The thought of a ton of brutalized remains in a corridor, stripped of flesh by sheer Scotland, appealed to me more than a bunch of gibs. It's also a lot less laggy than a bazillion gibs and bloody footprints everywhere. Same for beheading.

jughu commented:
Or both?

AnturK commented:
This done ?

Xhuis commented:
Yeah, it's ready.

AnturK commented:
Forgot to ask before but what's the source of the sound.

YakumoChen commented:
SCOTLAND FOREVER should be obligatory upon last man standing tbh

AnturK commented:
That was more question about the licenses on these specific versions.

Jordie0608 commented:
>two 30 second long 600kb sound files

I like bagpipes but geez why.

optimumtact commented:
B R A V E

Davdichan commented:
[url=https://www.youtube.com/watch?v=DkeGMaq0O98]HEEEEEEEEEEEEEERE WE AAAAARE! BOOOOORN TO BE KIIINGS! WE'RE THE PRINCES OF THE UNIVERSE![/url]
